### ISSUE TITLE
Segmentation faults with pynetcf

### ISSUE THREAD

**Issue by s-scherrer** [TUW-GEO/pynetcf#57]
In the `qa4sm_preprocessing` CI we get segmentation faults issued by this line:

https://github.com/TUW-GEO/pynetcf/blob/c634667a2b1edabd8710f8f61b7c3b70c72f42ee/src/pynetcf/base.py#L113

Unfortunately it is not possible to reproduce the issue locally for @wpreimes  or me. @pstradio recently encountered similar issues with "random" segfaults (reproducible, but the gpi at which it occurred depended on how exactly the program was run).

Does anyone have an idea what causes this and how to solve it?

**Comment by sebhahn** [comment]
if it is this line, the problem might be related to the underlying netCDF4 package version or the netCDF file itself

**Comment by s-scherrer** [comment]
I assume the problem is somehow related to opening it multiple times or calling `.read(gpi)` in a certain order, because opening the netCDF file directly without pynetcf is not a problem, and the problem also only occurs in some specific configurations which are hard to reproduce.

The underlying issue is probably in netCDF4, but I had hoped that there would be some workaround for `pynetcf`.

**Comment by sebhahn** [comment]
We would need more information on these configurations to find the problem

Have you tried the [Faulthandler](https://docs.python.org/3/library/faulthandler.html)?

Code usage:


```
import faulthandler

faulthandler.enable()

// bad code goes here
```

Shell usage:
```
$ python3 -q -X faulthandler
>>> /// bad cod goes here
```

**Comment by s-scherrer** [comment]
Thanks for the hint, @pstradio might be interesting for your issue.

In my case the issue occurs on the github CI but cannot be reproduced locally unfortunately, so there's not a lot more info I can give you. But I'll try to investigate a bit more and let you know if I find something.

**Comment by s-scherrer** [comment]
Thanks @sebhahn, I think this should be solved now via #58
